edx_api/grades/__init__.py

```diff
@@ -73,4 +73,34 @@ def get_student_current_grades(self, username, course_ids=None):
                 if error.response.status_code >= 500:
                     raise
 
-        return CurrentGrades(all_current_grades)
+        return CurrentGrades(all_current_grades, restrict_to="username")
+
+    def get_course_current_grades(self, course_id):
+        """
+        Returns a CurrentGrades object for all users in the specified course.
+
+        Args:
+            course_id (str): an edX course ids.
+
+        Returns:
+            CurrentGrades: object representing the student current grades
+
+        Authorization:
+            The authenticated user must have staff permissions to see grades for all users
+            in a course.
+        """
+        resp = self.requester.get(
+            urljoin(
+                self.base_url,
+                '/api/grades/v1/courses/{course_key}/'.format(
+                    course_key=course_id
+                )
+            )
+        )
+
+        resp.raise_for_status()
+
+        return CurrentGrades(
+            [CurrentGrade(entry) for entry in resp.json()],
```

Comment:
Shoot, so much for having consistent APIs. Apparently between Ironwood and Hawthorn the response of this API changed. It's now made paginated, so this doesn't work on current master because the response isn't a list of grades, it's a dictionary with `previous`, `next`, and `results` keys.

I got it to work by changing this to:
```
[CurrentGrade(entry) for entry in resp.json().get('results')],
```

Should, and could, we check the response and handle both possibilities? That way we could support Hawthorn and Ironwood.

Here's the commit that broke compatibility: https://github.com/edx/edx-platform/commit/b3c31535de276089e497b93a05d4ce1794a44b23

Reply:
Ah, that's a good idea.  I wish edx would maintain compatible api versioning.  This should have been put into a v2 api.

Reply:
But now we also have to deal with paginated responses.  I'll see if they've got a pattern for that in the api client already...

Reply:
Ohhh.  Maybe that's why they're fetching single responses, one by one instead of using the dedicated list endpoints.